packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -612,26 +612,32 @@ abstract class RouteTransitionRecord {
   /// Retrieves the wrapped [Route].
   Route<dynamic> get route;
 
-  /// Whether this route is entering the screen.
+  /// Whether this route is waiting for the decision on how to enter the screen.
   ///
   /// If this property is true, this route requires an explicit decision on how
   /// to transition into the screen. Such a decision should be made in the
   /// [TransitionDelegate.resolve].
-  bool get isEntering;
+  bool get isWaitingForEnteringDecision;
 
-  bool _debugWaitingForExitDecision = false;
+  /// Whether this route is waiting for the decision on how to exit the screen.
+  ///
+  /// If this property is true, this route requires an explicit decision on how
+  /// to transition off the screen. Such a decision should be made in the
+  /// [TransitionDelegate.resolve].
+  bool get isWaitingForExitingDecision => _isWaitingForExitingDecision;
+  bool _isWaitingForExitingDecision = false;
```

Comment:
slightly cleaner and to be symmetric to `isWaitingForEnteringDecision`: Move this field into _RouteEntry and add a method there (something like markNeedsExitingDecision) to set it to true.